module4-authentication-and-security/r1.1.1-authentication-persistence/README.md

```diff
@@ -0,0 +1,268 @@
+
+The only way to recover from this point is to generate a new Key (Key-pair) that will be used across systems from here on. This would mean that all the existing client tokens are invalidated, and every user would have to login again. Imagine if one day all Facebook users are suddenly logged out.
+
+### CON: JWTs expire at specific intervals
+
+When a JWT is created it is given a specific expiration instant. The life of a JWT is definitive and it is recommended that it is somewhat small (think minutes not hours).
+
+Compared with traditional sessions, JWTs are quite different. Sessions are always a specific duration from the last interaction with the user. This means that if the user clicks a button, their session is extended. If you think about most applications you use, this is pretty common. You are logged out of the application after a specific amount of inactivity.
+
+JWTs, on the other hand, are not extended on user interaction. Instead, they are programmatically replaced by creating a new JWT for the user.
+
+To solve this problem, most applications use refresh tokens. Refresh tokens are opaque tokens that are used to generate new JWTs. Refresh tokens also need to expire at some point, but they can be more flexible in this mechanism because they are persisted in the identity provider or database.
+
+### CON: Data Overhead
+
+The overall size of a JWT is quite more than that of a normal session token, which makes it longer whenever more data is added to it.
+
+So, if you’re adding more claims in the token, it will impact the overall loading speed and thus hamper the user experience.
+
+This situation can be fixed if the right development practices are followed and minimum but essential data is added to the JWT.
+
+### CON: JWTs aren’t easily revocable
+
+This means that a JWT could be valid even though the user’s account has been suspended or deleted. Some solutions around this are available but they mostly require trips to the identity provider or the database, which JWT is essentially developed to minimize.
+
+So if your app has the potential to deactivate or revoke user access frequently, think twice before using JWTs.
+
+## Conclusion
+
+In this lesson, we explored authentication persistence using both server-side sessions and client-side tokens. Each has its pros and cons. Sessions have been used for more than 2 decades now, and they are quite robust. However, their limitation comes from poor portability.
+
+JWTs, on the other hand, are also robust and pretty portable. However, they come with their overhead and access longevity issues.
+
+When building your app, always measure the pros and cons of both of these approaches and take your time to decide which one is best for you.
+
+## Read more
+
+- https://www.hebergementwebs.com/news/use-of-session-cookies-vs-jwt-for-authentication
```

Comment:
are these all related to JWTs? i think these references should be colocated to where they are. if its all about JWTs its a bit too many links, i dont think a student is going to click 6 links

Reply:
They are related to the whole topic, session, and JWTs. And I think they are helpful to include here.